modules/tensor_mechanics/src/materials/ADGeneralizedRadialReturnStressUpdate.C

```diff
@@ -178,3 +184,97 @@ ADGeneralizedRadialReturnStressUpdate::outputIterationSummary(std::stringstream
   }
   ADGeneralizedReturnMappingSolution::outputIterationSummary(iter_output, total_it);
 }
+
+void
+ADGeneralizedRadialReturnStressUpdate::rotateHillTensor(ADDenseMatrix & hill_tensor)
+{
+  _transformation_tensor.zero();
+
+  const Real s = std::sin(_angle(2) * libMesh::pi / 180.0);
+  const Real c = std::cos(_angle(2) * libMesh::pi / 180.0);
+
+  _transformation_tensor(0, 0) = c * c;
+  _transformation_tensor(0, 1) = s * s;
+  _transformation_tensor(0, 5) = 2 * c * s;
+
+  _transformation_tensor(1, 0) = s * s;
+  _transformation_tensor(1, 1) = c * c;
+  _transformation_tensor(1, 5) = -2 * c * s;
+
+  _transformation_tensor(2, 2) = 1.0;
+
+  _transformation_tensor(3, 3) = c;
+  _transformation_tensor(3, 4) = s;
+
+  _transformation_tensor(4, 3) = s;
+  _transformation_tensor(4, 4) = c;
+
+  _transformation_tensor(5, 0) = -c * s;
+  _transformation_tensor(5, 1) = c * s;
+  _transformation_tensor(5, 5) = c * c - s * s;
+
```

Comment:
Are these transformations about the Z axis? If so, you could use the `_z` suffix.

Reply:
Yes, this is about the z-axis. But I kept the name generic as it is the default material property. It saves us from having to define another dense matrix. I will add a comment to clarify.

Reply:
Sudipta, maybe we could stick to rotating the 6 coefficients directly and only store those in the material. That coefficient rotation could be a namespaced helper method.